src/api/Controllers/HydrotestController.cs

```diff
@@ -174,18 +183,37 @@ private bool AreHydrotestCodelistsCompatible(Hydrotest hydrotest)
             }
         }
 
-        var compatibleCodelistIds = new List<int>();
+        var kindCompatible = true;
+        var flowDirectionsCompatible = true;
+        var evaluationMethodsCompatible = true;
 
-        // If there are CodelistIds, find the compatible CodelistIds for the flow direction and evaluation method options.
-        if (hydrotest.CodelistIds?.Count > 0 && testKindGeolCodes.Count > 0)
+        var kindCodelistIds = hydrotestCodelists.Where(c => c.Schema == HydrogeologySchemas.HydrotestKindSchema).Select(c => c.Id).ToList();
+        kindCompatible = hydrotest.KindCodelistIds?.All(c => kindCodelistIds.Contains(c)) ?? true;
+
+        if (testKindGeolCodes.Count > 0)
         {
-            compatibleCodelistIds.AddRange(hydrotestKindCodelistIds);
-            compatibleCodelistIds.AddRange(GetCompatibleCodelistIds(testKindGeolCodes, HydrogeologySchemas.FlowdirectionSchema, HydroCodeLookup.HydrotestFlowDirectionOptions));
-            compatibleCodelistIds.AddRange(GetCompatibleCodelistIds(testKindGeolCodes, HydrogeologySchemas.EvaluationMethodSchema, HydroCodeLookup.HydrotestEvaluationMethodOptions));
+            // If there are FlowDirectionCodelistIds, check their compatibility with the test kind.
+            if (hydrotest.FlowDirectionCodelistIds.Count > 0)
+            {
+            var compatibleFlowDirectionCodelistIds = GetCompatibleCodelistIds(testKindGeolCodes, HydrogeologySchemas.FlowdirectionSchema, HydroCodeLookup.HydrotestFlowDirectionOptions)
+                .Union(hydrotestCodelists.Where(c => c.Schema == HydrogeologySchemas.FlowdirectionSchema).Select(c => c.Id))
+                .ToList();
+
+            flowDirectionsCompatible = hydrotest.FlowDirectionCodelistIds?.All(c => compatibleFlowDirectionCodelistIds.Contains(c)) ?? true;
```

Comment:
PP: Das kann vereinfacht ausgedrückt werden:
```suggestion
            flowDirectionsCompatible = hydrotest.FlowDirectionCodelistIds?.All(compatibleFlowDirectionCodelistIds.Contains) ?? true;
```